hare/metrics.go

```diff
@@ -45,4 +49,55 @@ var (
 		[]string{},
 		prometheus.ExponentialBuckets(4, 2, 3),
 	).WithLabelValues()
+
+	msgDelayTracker = newMessageDelayTracker()
 )
+
+func newMessageDelayTracker() *messageDelayTracker {
+	m := make(map[MessageType]map[bool]prometheus.Observer)
+	for _, t := range []MessageType{pre, status, proposal, commit, notify} {
+		m[t] = make(map[bool]prometheus.Observer)
+		for _, pos := range []bool{true, false} {
+			sign := "pos"
+			if !pos {
+				sign = "neg"
+			}
+
+			m[t][pos] = metrics.NewHistogramWithBuckets(
+				fmt.Sprintf("%v_%v_delay_seconds", strings.ToLower(t.String()), sign),
+				namespace,
+				fmt.Sprintf("delta between expected time of receipt of %v message and actual receipt time", t),
+				[]string{},
+				prometheus.ExponentialBuckets(0.2, 2, 10),
```

Comment:
i opted for ms for similar thing, do you think that working with floats is better?

Reply:
I was thinking of using ms too but then I read in the [prometheus label naming practices](https://prometheus.io/docs/practices/naming/) that you should use base units, which seemed a bit arbitrary to me but I switched :) I don't have any preference for float or non float, but in the end `Observe` expects a float, so I don't see any problem with using floats. Is there a problem?

Reply:
Happy to switch back to ms if that's going to be more inline with our current metrics

Reply:
i guess seconds might be somewhat easier to consume, with milliseconds it looks like this

![image](https://user-images.githubusercontent.com/1964576/225874466-0a1b61e2-a31f-4dbb-b790-dd9c6cae33ac.png)